homeassistant/components/smtp/notify.py

```diff
@@ -241,13 +247,21 @@ def _build_text_msg(message):
     return MIMEText(message)
 
 
-def _attach_file(atch_name, content_id=""):
+def _attach_file(hass, atch_name, content_id=""):
     """Create a message attachment.
 
     If MIMEImage is successful and content_id is passed (HTML), add images in-line.
     Otherwise add them as attachments.
     """
     try:
+        file_path = Path(atch_name).parent
+        if not hass.config.is_allowed_path(str(file_path)):
+            _LOGGER.warning(
```

Comment:
We should raise `ServiceValidationError`.

Reply:
https://github.com/home-assistant/core/pull/104981